internal/serviceset/util.go

```diff
@@ -376,32 +376,8 @@ func needsUpdate(
 	providerSpec kcmv1.StateManagementProviderConfig,
 	services []kcmv1.Service,
 ) bool {
-	// we'll need to update provider configuration if it was changed.
-	if !equality.Semantic.DeepEqual(providerSpec, serviceSet.Spec.Provider) {
-		return true
-	}
```

Comment:
Removed this check for the following reasons:

- We don't need it anymore because `provider.name` is [immutable](https://github.com/k0rdent/kcm/blob/eea7692411036fd80418ffecd46656dfcbe60e28/api/v1beta1/serviceset_types.go#L118-L121).
- This check was masking the bug mentioned in comment below